load_balance.go

```diff
@@ -348,14 +357,26 @@ func getHostWithLeastConns(li *ClusterLoadInfo) *lbHost {
 	leastCnt := int(math.MaxInt64)
 	leastLoaded := ""
 	if li.config.topologyKeys != nil {
-		for _, tk := range li.config.topologyKeys {
-			for _, h := range li.zoneList[tk] {
+		for i := 0; i < len(li.config.topologyKeys); i++ {
+			var servers []string
+			for _, tk := range li.config.topologyKeys[i] {
+				ToCheckStar := strings.Split(tk, ".")
+				if ToCheckStar[2] == "*" {
+					tk = ToCheckStar[0] + "." + ToCheckStar[1]
+				}
+				servers = append(servers, li.zoneList[tk]...)
+			}
+			for _, h := range servers {
 				if !isHostAway(li, h) && li.hostLoad[h] < leastCnt {
 					leastLoaded, leastCnt = h, li.hostLoad[h]
 				}
 			}
+			if leastCnt != int(math.MaxInt64) && leastLoaded != "" {
```

Comment:
This may break from the for loop without going through all the topology-keys specified. But that is not intended, right?

Reply:
It will break from for loop if it finds a least loaded active server in the Primary placement zone(i=0) else it will go to i=1 i.e first fallback and will continue so on.